layers/ntp.go

```diff
@@ -282,42 +221,53 @@ const ntp_minimum_record_size_in_bytes int = 48
 // represents in a structured form the NTP record present as the UDP
 // payload in an NTP UDP packet.
 //
+
+type NTPLeapIndicator uint8
+type NTPVersion uint8
+type NTPMode uint8
+type NTPStratum uint8
+type NTPPoll int8
+type NTPPrecision int8
+type NTPRootDelay uint32
+type NTPRootDispersion uint32
+type NTPReferenceID uint32
+type NTPReferenceTimestamp uint64
+type NTPOriginTimestamp uint64
+type NTPReceiveTimestamp uint64
+type NTPTransmitTimestamp uint64
+
 type NTP struct {
 	BaseLayer // Stores the packet bytes and payload bytes.
 
-	LeapIndicator uint8 // [0,3]. Indicates whether leap second(s) is to be added.
-	Version       uint8 // [0,7]. Version of the NTP protocol.
-	Mode          uint8 // [0,7]. Mode.
-	Stratum       uint8 // [0,255]. Stratum of time server in the server tree.
-	Poll          int8  // [-128,127]. The maximum interval between
-	//             successive messages, in log2 seconds.
-	Precision int8 // [-128,127]. The precision of the system clock,
-	//             in log2 seconds.
-	RootDelay uint32 // [0,2^32-1]. Total round trip delay to the reference clock
-	//             in seconds times 2^16.
-	RootDispersion uint32 // [0,2^32-1]. Total dispersion to the reference clock,
-	//             in seconds times 2^16.
-	ReferenceID        uint32 // ID code of reference clock [0,2^32-1].
-	ReferenceTimestamp uint64 // Most recent timestamp from the reference clock.
-	OriginTimestamp    uint64 // Local time when request was sent from local host.
-	ReceiveTimestamp   uint64 // Local time (on server) that request arrived at server host.
-	TransmitTimestamp  uint64 // Local time (on server) that request departed server host.
+	LeapIndicator      NTPLeapIndicator      // [0,3]. Indicates whether leap second(s) is to be added.
+	Version            NTPVersion            // [0,7]. Version of the NTP protocol.
+	Mode               NTPMode               // [0,7]. Mode.
+	Stratum            NTPStratum            // [0,255]. Stratum of time server in the server tree.
+	Poll               NTPPoll               // [-128,127]. The maximum interval between successive messages, in log2 seconds.
+	Precision          NTPPrecision          // [-128,127]. The precision of the system clock, in log2 seconds.
+	RootDelay          NTPRootDelay          // [0,2^32-1]. Total round trip delay to the reference clock in seconds times 2^16.
```

Comment:
If the values contain the same type of data (like Poll/Precision and RootDelay/RootDispersion), feel free to make those the same type.

Also, *Timestamp can probably all be a NTPTimestamp.